### PR TITLE
Fix #1235 ViewsOpenResponse type is incorrect

### LINKED ISSUE #1235: `ViewsOpenResponse` type is incorrect
### Description

the type is defined as 

```ts
export declare type ViewsOpenResponse = WebAPICallResult & {
    ok?: boolean;
    error?: string;
    response_metadata?: ResponseMetadata;
    needed?: string;
    provided?: string;
};
```

however in actuality, it has an additional `view` property. Here is an example opened view response:

```json
{
   "ok":true,
   "view":{
      "id":"V021U522M6G",
      "team_id":"OMITTED",
      "type":"modal",
      "blocks":[
         {
            "type":"section",
            "block_id":"JtAZ+",
            "text":{
               "type":"mrkdwn",
               "text":"Loading...",
               "verbatim":false
            }
         }
      ],
      "private_metadata":"",
      "callback_id":"CONFIRM_MESSAGE_SELECTION",
      "state":{
         "values":{
            
         }
      },
      "hash":"1620800476.cy2CiRSe",
      "title":{
         "type":"plain_text",
         "text":"Create a new issue",
         "emoji":true
      },
      "clear_on_close":false,
      "notify_on_close":false,
      "close":{
         "type":"plain_text",
         "text":"Cancel",
         "emoji":true
      },
      "submit":null,
      "previous_view_id":null,
      "root_view_id":"V021U522M6G",
      "app_id":"OMITTED",
      "external_id":"",
      "app_installed_team_id":"OMITTED",
      "bot_id":"OMITTED"
   },
   "response_metadata":{
      "scopes":[
      ]
   }
}
```

I suspect the same bug is present in other view related response types.

### What type of issue is this? (place an `x` in one of the `[ ]`)
- [x] bug
- [ ] enhancement (feature request)
- [ ] question
- [ ] documentation related
- [ ] testing related
- [ ] discussion

### Requirements (place an `x` in each of the `[ ]`)
* [x] I've read and understood the [Contributing guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've searched for any related issues and avoided creating a duplicate issue.

---

### Bug Report

Filling out the following details about bugs will help us solve your issue sooner.

#### Packages:

Select all that apply:

- [x] `@slack/web-api`
- [ ] `@slack/events-api`
- [ ] `@slack/interactive-messages`
- [ ] `@slack/rtm-api`
- [ ] `@slack/webhooks`
- [ ] `@slack/oauth`
- [ ] `@slack/socket-mode`
- [ ] I don't know

#### Reproducible in:

package version: 6.2.3

node version: 12.19.0

OS version(s): N/A

#### Steps to reproduce:

1. Invoke `client.views.open(options)`

#### Expected result:

The response type corresponds to what properties actually exist. As far as I can tell, it is just missing the `view` property, which is quite important.

#### Actual result:

There is no such property defined on the interface.

### PR DESCRIPTION
**Pull request by seratch**
###  Summary

This pull request fixes #1235 by updating the API response types. see also: https://github.com/slackapi/java-slack-sdk/pull/743

### Requirements (place an `x` in each `[ ]`)

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/node-slack-sdk/blob/master/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).

### CODE REVIEW

File: packages/web-api/src/response/ViewsOpenResponse.ts [hunk 1]
```diff
@@ -2,12 +2,141 @@
...
+
+export interface Close {
+  type?:     Type;
+  text?:     string;
+  emoji?:    boolean;
+  verbatim?: boolean;
+}
+
+export enum Type {
+  Empty = '',
```
Comment (seratch) [inline packages/web-api/src/response/ViewsOpenResponse.ts:120]: This looks invalid - I will look into this later on.

File: packages/web-api/src/response/ViewsPublishResponse.ts [hunk 2]
```diff
@@ -2,12 +2,141 @@
...
 export interface ResponseMetadata {
   messages?: string[];
 }
+
+export interface View {
+  id?:                    string;
+  team_id?:               string;
+  type?:                  string;
+  title?:                 Close;
+  submit?:                Close;
```
Comment (rr-codes) [inline packages/web-api/src/response/ViewsPublishResponse.ts:22]: I understand type wise that `title`, `submit`, and `close` are the same (I think so at least), but semantically, it seems really weird to have the types of `title` and `submit` be `Close`

File: packages/web-api/src/response/ViewsPublishResponse.ts [hunk 3]
```diff
@@ -2,12 +2,141 @@
...
+  view?:              View;
+  response_metadata?: ResponseMetadata;
 };
 
 export interface ResponseMetadata {
   messages?: string[];
 }
+
+export interface View {
+  id?:                    string;
```
Comment (rr-codes) [inline packages/web-api/src/response/ViewsPublishResponse.ts:18]: also, is there a reason why every single property is optional, even though this is not the case? `id` for example should always be present

File: packages/web-api/src/response/ViewsPushResponse.ts [hunk 4]
```diff
@@ -2,12 +2,141 @@
...
+  trigger_actions_on?: string[];
+}
+
+export interface Filter {
+  exclude_external_shared_channels?: boolean;
+  exclude_bot_users?:                boolean;
+}
+
+export interface Option {
+  text?:        Close;
```
Comment (rr-codes) [inline packages/web-api/src/response/ViewsPushResponse.ts:135]: ya again this type semantically shouldn't be `Close`

File: packages/web-api/src/response/ViewsOpenResponse.ts [hunk 5]
```diff
@@ -2,12 +2,141 @@
...
+  image_url?:    string;
+  alt_text?:     string;
+  fallback?:     string;
+  image_width?:  number;
+  image_height?: number;
+  image_bytes?:  number;
+}
+
+export interface Element {
+  type?:                            string;
```
Comment (rr-codes) [inline packages/web-api/src/response/ViewsOpenResponse.ts:72]: `type` property values are always defined as far as I'm aware

Comment (seratch) [reply packages/web-api/src/response/ViewsOpenResponse.ts:72]: @rr-codes Thanks for the comment. You're right. This `type` and some others are not optional but as we generate the types using code generator (quicktype), this is not customizable.

File: packages/web-api/src/response/ViewsPublishResponse.ts (hunk 3 above)
Comment (seratch) [reply packages/web-api/src/response/ViewsPublishResponse.ts:18]: the same as my above comment

File: packages/web-api/src/response/ViewsPublishResponse.ts (hunk 2 above)
Comment (seratch) [reply packages/web-api/src/response/ViewsPublishResponse.ts:22]: This is automatically determined by the generator tool. Unfortunately, there is no way to customize the name,

File: packages/web-api/src/response/ViewsPushResponse.ts (hunk 4 above)
Comment (seratch) [reply packages/web-api/src/response/ViewsPushResponse.ts:135]: same as above

**Review by seratch — COMMENTED** [review]
@rr-codes Thanks for your comment (I really appreciate it). However, as we use code generator, all the points you mentioned are not customizable. We've decided to go with these limitations.

File: packages/web-api/src/response/ViewsOpenResponse.ts (hunk 1 above)
Comment (seratch) [reply packages/web-api/src/response/ViewsOpenResponse.ts:120]: I've looked into this and updating this is not so simple. Let us leave it as-is this time.

**Comment by rr-codes** [comment]
@seratch out of curiosity, what is the generator that's being used to generate the types? Is it open source?

**Comment by seratch** [comment]
@rr-codes As I mentioned above, we are using https://github.com/quicktype/quicktype for it.

**Comment by rr-codes** [comment]
@seratch oh whoops completely missed that aha, thanks so much!!

**Comment by rr-codes** [comment]
Also sorry @seratch one last question, whats the actual workflow of using Prototype for the types? Does it use [this schema ](https://raw.githubusercontent.com/slackapi/slack-api-specs/master/web-api/slack_web_openapi_v2_without_examples.json) with the json schema option?

**Comment by seratch** [comment]
@rr-codes Ideally, we should use the slack-api-specs but we don't use it. The main reason is that the outputs in the project are incomplete and are not up-to-date. We use the JSON data generated by the official Java SDK. Refer to https://github.com/slackapi/node-slack-sdk/pull/1188 for details.

**Review by misscoded — APPROVED** [review]

**Pull request merged**